third_party/terraform/resources/resource_sql_database_instance.go

```diff
@@ -452,6 +451,13 @@ func isFirstGen(d *schema.ResourceData) bool {
 	return !regexp.MustCompile("db*").Match([]byte(tier))
 }
 
+func isSqlOperationInProgressError(err error) (bool, string) {
```

Comment:
We're catching multiple types of 409 here, not just in progress, right?

Reply:
i think all the 409s that come out of SQL have operation in progress messages based on isRetryableError which has overlapping functionality. We can certainly add that error message, but since some of the code in this file was previously only using error code as a check I don't want to break it by accident. I think our eventual goal would be to move the 409 + "operationInProgress" check out of isRetryableError and into cloud SQL resources as error predicates because I'm prettyyy sure only SQL has this problem (but I also don't want to do that in this PR because maybe some other API is using 409s for retryable errors?)

Reply:
Won't a 409 conflict also occur when we try to reuse a reserved name? That's my impression based on "Error, failed to create instance %s with error code 409: %s. This may be due to a name collision - SQL instance names cannot be reused within a week.".